client/app/components/game-client.js

```diff
@@ -0,0 +1,64 @@
+import Ember from 'ember';
+import GameChannel from 'vilive/utils/game-channel';
+
+export default Ember.Component.extend({
+  init() {
+    this.set('messages', Ember.makeArray());
+    GameChannel.init(this.get('session-manager').currentToken());
+    GameChannel.on('event', payload => {
+      this.putMessage(true, payload.body);
+      this.scrollToBottomMessages();
+    });
+    this._super(...arguments);
+  },
+
+  actions: {
+    sendCommand() {
+      const command = this.get('command');
+
+      if (!command || command.length === 0) { return; }
+
+      // hack for template at the moment
+      // false means cliean, true means server
+      this.putMessage(false, command);
+      this.scrollToBottomMessages();
+
+      GameChannel.push('user_cmd', {body: command});
+      this.set('command', null);
+    }
+  },
+
+  /**
+    This does not fire after each message is inserted just when the component
+    is inserted, in here we are going to focus on the text box.
+  */
+  didInsertElement() {
+    this.$('#command-input').focus();
+  },
+
+  /**
+    Need to scroll the div after the message has been rendered.
+    otherwise it stops short and scroll to last message inserted
+  */
+  scrollToBottomMessages() {
+    const div = this.$('.panel-body');
+    Ember.run.schedule('afterRender', () => {
+      div.scrollTop(div.prop("scrollHeight") - div.prop("clientHeight"));
+    });
+  },
+
+  putMessage(isServer, msg) {
+    const dt = new Date();
+    const date = dt.toLocaleDateString();
+    const time = dt.toLocaleTimeString();
+    this.get('messages').pushObject({
+      isServer: isServer,
+      timeStamp: `${date} ${time}`,
```

Comment:
Yeah totally agree, this is just copying what we had - the infrastructure isn't there yet for that.

In fact if we can get this merged up I would like to start creating issues around these so they don't get lost and there are a few other things that need to be ironed out like what all is going to be included in a the payload etc.